### PR TITLE
feat(metis): add domains&costobject endpoint handler

### PR DESCRIPTION
**Pull request by IvoGoman**
- remove unused code from projects endpoint
- fix some copy+paste errors
- add domains get and list
- add costobjects get and list

### CODE REVIEW

File: metis/v1/identity/domains/model.go [hunk 1]
```diff
@@ -0,0 +1,17 @@
+package domains
```
[the remark this answers is not in the document]
Comment (IvoGoman) [reply metis/v1/identity/domains/model.go]: True, this aligns better with the other clients.

File: metis/v1/identity/costobjects/results.go [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+}
+
+// Extract accepts a Page struct, specifically an v1.CommonPage struct,
+// and extracts the elements into a slice of CostObjects structs.
+func Extract(r pagination.Page) ([]CostObject, error) {
+	var s struct {
+		CostObjects []CostObject `json:"items"`
+	}
+	if err := (r.(v1.CommonPage)).ExtractInto(&s); err != nil {
+		return []CostObject{}, err
```
Comment (kayrus) [inline metis/v1/identity/costobjects/results.go:36]: is there a reason to return non-nil empty object?

File: metis/v1/identity/costobjects/results.go [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+		} `json:"data"`
+	}
+	err := r.ExtractInto(&s)
+	if err != nil {
+		return nil, err
+	}
+	if s.Data.Items == nil {
+		return nil, nil
+	}
+	return &s.Data.Items[0], nil
```
Comment (kayrus) [inline metis/v1/identity/costobjects/results.go:62]: is there a chance that `items` array can be longer than 1 item?

File: metis/v1/identity/costobjects/results.go (hunk 2 above)
Comment (IvoGoman) [reply metis/v1/identity/costobjects/results.go]: No reason, have changed it for all Extract methods.

File: metis/v1/identity/costobjects/results.go (hunk 3 above)
Comment (IvoGoman) [reply metis/v1/identity/costobjects/results.go:62]: The API will always return an Array of Items. The Get is always performed for a certain ID and will return at most one result.

Comment (kayrus) [reply metis/v1/identity/costobjects/results.go:62]: that's questionable  API design, but if it works this way, then ok.

**Review by kayrus — APPROVED** [review]

File: metis/v1/identity/costobjects/results.go (hunk 3 above)
Comment (IvoGoman) [reply metis/v1/identity/costobjects/results.go]: Agreed, since these are new endpoints which are not yet used I have changed the API.

**Pull request merged**